### PR TITLE
CXX-886 Implement lookup operator for aggregation

### PR DESCRIPTION
**Pull request by samantharitter**
@xdg PTAL?

### CODE REVIEW

File: src/mongocxx/pipeline.hpp [hunk 1]
```diff
@@ -80,6 +80,20 @@ class MONGOCXX_API pipeline {
     pipeline& limit(std::int32_t limit);
 
     ///
+    /// Performs a left outer join to an unsharded collection in the same database to filter in
+    /// documents from the “joined” collection for processing.
```
Comment (Machyne) [inline src/mongocxx/pipeline.hpp]: I don't know if we care, but these quotes look non-ASCII to me. Does that matter?

File: src/mongocxx/test/collection.cpp [hunk 2]
```diff
@@ -532,6 +532,69 @@ TEST_CASE("CRUD functionality", "[driver::collection]") {
...
     }
 
+    SECTION("aggregation $lookup operator", "[collection]") {
+        auto people_coll_name = "people_on_the_block";
+        auto people_coll = db.create_collection(people_coll_name);
+        auto houses_coll_name = "houses_on_the_block";
+        auto houses_coll = db.create_collection(houses_coll_name);
+
+        // populate one collection with names
+        document name1;
```
Comment (hanumantmk) [inline src/mongocxx/test/collection.cpp]: You don't have to rewrite it, but more stylish:

auto name1 = document{} << "key" << "value" << finalize;

File: src/mongocxx/pipeline.hpp (hunk 1 above)
Comment (hanumantmk) [inline src/mongocxx/pipeline.hpp]: Yeah, those are unicode (bytes):

�       226
�       128
�       156
j       106
o       111
i       105
n       110
e       101
d       100
�       226
�       128
�       157

Let's make those ascii

**Comment by hanumantmk** [comment]
LGTM % nit

**Comment by acmorrow** [comment]
LGTM as well, same nit. Any idea how those unicode quotes got in there?

**Comment by samantharitter** [comment]
Weird!  Possibly from copying and pasting from our online documentation about $lookup?  Fixed!

**Comment by Machyne** [comment]
^ definitely a copy paste from the docs [here](https://docs.mongodb.org/manual/reference/operator/aggregation/lookup/#pipe._S_lookup)

From the page source:

```
<p>Performs a left outer join to an unsharded collection in the <em>same</em>
database to filter in documents from the &#8220;joined&#8221; collection for
```

Which (thanks to markdown support of html) ends up looking like:
> <p>Performs a left outer join to an unsharded collection in the <em>same</em>
> database to filter in documents from the &#8220;joined&#8221; collection for

Crazy!

**Pull request merged**